amr-wind/wind_energy/actuator/disk/uniform_ct_ops.H

```diff
@@ -0,0 +1,201 @@
+
+void write_netcdf(
+    const std::string&,
+    const UniformCtData&,
+    const ActInfo&,
+    const ActGrid&,
+    const amrex::Real);
+} // namespace disk
+
+namespace ops {
+
+void collect_parse_conflicts(
+    const utils::ActParser& pp,
+    const std::string& p1,
+    const std::string& p2,
+    std::ostringstream& ss);
+void collect_parse_dependencies(
+    const utils::ActParser& pp,
+    const std::string& p1,
+    const std::string& p2,
+    std::ostringstream& ss);
+
+void required_parameters(UniformCt::MetaType& meta, const utils::ActParser& pp);
+
+void optional_parameters(UniformCt::MetaType& meta, const utils::ActParser& pp);
+
+void check_for_parse_conflicts(const utils::ActParser& pp);
+
+void compute_and_normalize_coplanar_vector(UniformCt::MetaType& meta);
+
+void final_checks(const UniformCt::MetaType& meta);
+
+amrex::RealBox compute_bounding_box(const UniformCt::MetaType& meta);
+
+void do_parse_based_computations(UniformCt::DataType& data);
+
+void compute_disk_points(
+    UniformCt::MetaType& meta,
+    VecList& points,
+    const vs::Vector& cylAxis,
+    const int offset,
+    const double dOffset);
+
+template <>
+struct ReadInputsOp<UniformCt, ActSrcDisk>
+{
+    void operator()(UniformCt::DataType& data, const utils::ActParser& pp)
+    {
+        // TODO get density at probe location
+        check_for_parse_conflicts(pp);
+        auto& meta = data.meta();
+        required_parameters(meta, pp);
+        optional_parameters(meta, pp);
+        final_checks(meta);
+        do_parse_based_computations(data);
+    }
+};
+
+template <>
+struct InitDataOp<UniformCt, ActSrcDisk>
+{
+    void operator()(UniformCt::DataType& data)
+    {
+        auto& grid = data.grid();
+        auto& meta = data.meta();
+
+        // only resize the members we are going to use
+        grid.pos.resize(meta.num_force_pts);
+        grid.force.resize(meta.num_force_pts);
+        grid.vel.resize(meta.num_vel_pts);
+        grid.vel_pos.resize(meta.num_vel_pts);
+
+        const auto& cVec = meta.coplanar_vec;
+        const auto& sVec = meta.sample_vec;
+        const auto& nVec = meta.normal_vec;
+
+        const auto& cc = meta.center;
+        {
+            const auto& dr = meta.dr;
+            for (int i = 0; i < meta.num_force_pts; ++i) {
+                grid.pos[i] = cc + (i + 0.5) * dr * cVec;
+            }
+        }
+        compute_disk_points(
+            meta, grid.vel_pos, sVec, 0, meta.diameters_to_sample);
+        compute_disk_points(meta, grid.vel_pos, nVec, meta.num_vel_pts / 2, 0);
+    }
+};
+
+template <>
+struct UpdateVelOp<UniformCt, ActSrcDisk>
+{
+    void operator()(UniformCt::DataType& data)
+    {
+        auto& meta = data.meta();
+        const auto& grid = data.grid();
+        auto& refVel = meta.reference_velocity;
+        auto& diskVel = meta.disk_velocity;
+        const int np = meta.num_vel_pts / 2;
+        for (int i = 0; i < np; i++) {
```

Comment:
Should `refVel` and `diskVel` be initialized to zero before entering the loop?  I'm getting some strange output velocities when running the disk, but not sure if that's the cause of this.

Reply:
:+1: update coming :)